src/rhsm/profile.py

```diff
@@ -43,16 +45,54 @@ class InvalidProfileType(Exception):
     pass
 
 
-class ModuleProfile(object):
+class ModulesProfile(object):
+
+    def __init__(self):
+        self.content = self.__generate()
+
+    def __str__(self):
+        return str(self.content)
+
+    def __eq__(self, other):
+        return self.content == other.content
+
+    @staticmethod
+    def __generate():
+        module_list = []
+        if dnf is not None:
+            base = dnf.Base()
+            base.read_all_repos()
+            base.fill_sack()
+            # FIXME: DNF doesn't provide public API for modulemd
+            try:
+                modules = base._moduleContainer
+            except AttributeError:
+                log.info("DNF does not provide modulemd functionality")
+                return []
+            all_module_list = modules.getModulePackages()
```

Comment:
When I run the code, I always get the `AttributeError` exception, because the `_moduleContainer` field doesn't exist. Is this always going to be the case? How can I get to the next block, where the `_moduleContainer` actually is there?

Reply:
Let's look at my reply to your previous comment. Master branch of libdnf provides this attribute.